File: swing/src/main/java/wannabe/swing/renderer/SwingRenderer.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

/** Displays a voxel. */
public abstract class SwingRenderer {
  abstract public void draw(Graphics g, Projected r);

  @Override public String toString() {
    return getClass().getSimpleName();
  }

  @Override public boolean equals(Object obj) {
    if (obj == null) return false;
    return getClass().equals(obj.getClass());
  }

  @Override public int hashCode() {
    return getClass().hashCode();
  }
}

File: swing/src/main/java/wannabe/swing/renderer/SwingRenderer.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

/** Displays a voxel in a swing UI.  */
abstract class SwingRenderer {
  abstract fun draw(
    g: Graphics,
    p: SwingProjected
  )

  override fun toString(): String = javaClass.simpleName

  override fun equals(other: Any?) = other != null && javaClass == other.javaClass

  override fun hashCode(): Int = javaClass.hashCode()
}
